### PR TITLE
Add source links to creator pages

### LINKED ISSUE #3496: Link to source on creator view
## Description

It might be nice to link the source on the creator view, given that we have that information and are displaying it anyway.

**Example**:

1. Visit https://staging.openverse.org/preferences and enable `additional_search_views`
2. Visit https://staging.openverse.org/image/source/smithsonian_air_and_space_museum/creator/Unknown/
3. The source name should be clicked on too

## Alternatives

Not sure if this is necessary or ideal, but just something I thought while navigating through the pages.

## Additional context

Related to #410

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #3496 by @AetherUnbound 

## Description

This PR updates the collection header component to match the updated designs.

The creator page has a scrollable area with a button linking to the source page in Openverse.

On narrow screens, the icons next to the `h1` heading are replaced with the words. To make this heading accessible, I wrapped the `h1` around the icon/collection name, as well as the name of the tag/source/creator. This is not perfect, especially for i18n, but it gives the screen reader user a better context.

The newly-added source button is scrollable to allow for long source names. To implement this, I extracted the scrolling features from `VByLine` into a separate component, and reused it for this button as well. Some of the extra margin/padding values were added to keep the height of the button correct while also making the focus ring visible (this is a problem for buttons in containers that have overflow set to anything other than visible.

~~This PR adds a link to the creator view results label. Due to limitations in how v.8 vue-i18n's `i18n` functional component works, we cannot use it with the pluralization settings (we need pluralization because the label has "x results"). This is why I didn't use `<i18n path="the creator label">` but created a workaround splitting the creator result label by the source name instead.~~

## Testing Instructions

Go to `/preferences` page and make sure the `additional_search_views` switch is `on`.
Go to a single result page, and click on a creator link button underneath the image or audio. On the source view, you should see the link to a source with an "external link" icon next to it. When you click it, you should see an analytics event logged in the console (and should see the source page open in a new tab).

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/components/VCollectionHeader/VCollectionHeader.vue [hunk 1]
```diff
@@ -167,6 +192,15 @@ export default defineComponent({
...
 <style scoped>
+@keyframes shimmer {
+  0% {
+    background-position: -1000px 0;
+  }
+  100% {
+    background-position: 1000px 0;
+  }
+}
+
```
Comment (obulat) [inline frontend/src/components/VCollectionHeader/VCollectionHeader.vue:203]:
```suggestion
```

**Review by AetherUnbound — DISMISSED** [review]
Oops, that's not quite what I had in mind, sorry I should have been more explicit! I was thinking of linking to _our own source view_, using the link with the little institution icon we have 😮 What do you think?

**Comment by obulat** [comment]
> Oops, that's not quite what I had in mind, sorry I should have been more explicit! I was thinking of linking to _our own source view_, using the link with the little institution icon we have 😮 What do you think?

@fcoveram, this was how I first understood this suggestion, and then I saw the mockup and added the link to the source site. 
What do you think, should we add the link button to the Openverse source page (i.e., https://openverse.org/source/flickr instead of flickr.com) to the creator page?

**Comment by fcoveram** [comment]
Oh, now I understand. And yes, it makes more sense linking to the source view as this page has a button to the external site on the far right side. I considered that option but read the sentence twice and understood we wanted to link the external source, but jumping between Openverse views make more sense.

I tried a few things that tap into a spacing improvement suggestion I shared time ago (in a ticket I can not find), and it looks like this.

Mockups of additional views on desktop

Mockups of additional views on mobile

Adding the small button next to results label on creator page requires a spacing update in the three views on desktop, but the change is minor and only impacts the results label area. But on small breakpoints, the changes involve more elements as the header variants (source, creator, and tag) initially designed have a `width: auto` button and the page type icon that make the layout more complex once small button is placed in the results label area.

I do not want to creep the PR scope, but worth discussing how this proposal feels to us, and if we agree on going with it, what changes belong to this ticket scope.

**Comment by AetherUnbound** [comment]
That design for the wider viewports was exactly what I had in mind Francisco! On the smaller breakpoints, do you think it would make sense to break the source over to a new line, separate from the results count text?

**Comment by fcoveram** [comment]
You are right @AetherUnbound, I don't know why I didn't consider that option.

Here is the new version and other spacing tweaks to the rest of views to keep consistency. Note that single result changes the small button style and one padding-left value.

Mockups of additional search views and single result view on mobile

Curious to hear @obulat's thought on this idea.

**Comment by obulat** [comment]
> Here is the new version and other spacing tweaks to the rest of views to keep consistency. Note that single result changes the small button style and one padding-left value.

I like the latest version where the source link is on a separate line. However, still having it as a sentence, split into 2 lines, might be a problem for accessibility because the sentence will be split into two, and a user who has a screen reader might not understand what it means.

Did we replace the "tag"/"source"/"creator" icon with the word on mobile?

**Comment by fcoveram** [comment]
> Still having it as a sentence, split into 2 lines, might be a problem for accessibility because the sentence will be split into two, and a user who has a screen reader might not understand what it means.

True, thanks for noticing it. I think we can change the copy to something like 
* [number] images by this creator in the following source
* [number] images by this creator in this source

> Did we replace the "tag"/"source"/"creator" icon with the word on mobile?

Yes. And in the single result page, the small buttons change from `filled: gray` to `transparent: gray` plus other spacing values in the same area.

**Comment by fcoveram** [comment]
I updated [the mockups](https://www.figma.com/file/niWnCgB7K0Y4e4mgxMrnRC/Additional-search-views?type=design&node-id=22-13656&mode=dev) and added dev notes.

**Review by fcoveram — CHANGES_REQUESTED** [review]
The changes look great. I have just three comments:

### Small breakpoints
* The external link in Creator and Source pages needs to be full-width.
* The results label in Tag and Source pages needs to have a padding bottom of `8px` to reach the the total spacing of `16px`. Figma dev notes show this more clearly and [this](https://share.cleanshot.com/YGnHBBnm) is the element I'm referring to.

### Bigger breakpoints
* The results label in Tag page needs to be inside an area of `32px` height. Figma dev notes show this more clearly.

**Review by zackkrida — APPROVED** [review]
Very nice; lgtm. It is behaving correctly in multiple browsers in MacOS and Linux on my local machines. Tested in Firefox, Safari, and Chrome.

**Comment by fcoveram** [comment]
I checked the last changes and it looks perfect ✨

**Review by fcoveram — APPROVED** [review]
Looks great ✨ 🚀

**Pull request merged**